File: app_kharid/databaseMaker/createDatabase.py
import pyodbc
import sqlite3
import ORM
from time import sleep
# Connect to the Access database
path = r'C:\\Users\\a\Desktop\\safar\\database.mdb;'
db = r'DRIVER={Microsoft Access Driver (*.mdb, *.accdb)};DBQ=' + path

conn = pyodbc.connect(db)
access_cursor = conn.cursor()

access_cursor.execute('SELECT * FROM Foroush_Detail')
rows = access_cursor.fetchall()

all = rows[::-1]

for row in all:
    print(row[0])
    try:
        ORM.create(
            KharidarName=row[18],
            KharidarLastNameSherkatName=row[19],
            KharidarNationalCode=row[21],
            HCKharidarTypeCode=row[13],
        )
    except:
        pass
    
access_cursor.close()
conn.close()

File: app_forush/databaseMaker/createDatabase.js
const ADODB = require("node-adodb");
const { Sequelize, DataTypes } = require("sequelize");

async function run() {
  // Connect to the Access database
  const path = process.cwd() + "/database.mdb";
  const connStr = `Provider=Microsoft.ACE.OLEDB.12.0;Data Source=${path};`;

  const connection = ADODB.open(connStr);

  // Define the model for the table
  const sequelize = new Sequelize({
    dialect: "sqlite",
    storage: "mydatabase.db",
  });

  const MyTable = sequelize.define(
    "companies",
    {
      KharidarName: DataTypes.STRING,
      KharidarLastNameSherkatName: DataTypes.STRING,
      KharidarNationalCode: DataTypes.STRING,
      HCKharidarTypeCode: DataTypes.STRING,
    },
    { timestamps: false }
  );

  // Retrieve the rows from the table
  const rows = await connection.query("SELECT * FROM Foroush_Detail");

  // Reverse the rows
  const all = rows.reverse();

  // Insert each row into the ORM
  for (const row of all) {
    console.log(row.Radif);
    try {
      await MyTable.create({
        KharidarName: row.KharidarName,
        KharidarLastNameSherkatName: row.KharidarLastNameSherkatName,
        KharidarNationalCode: row.KharidarNationalCode,
        HCKharidarTypeCode: row.HCKharidarTypeCode,
      });
    } catch (err) {}
  }

  // Close the database connection
  connection.close();
}

run();
